src/src/InputConfigImplementation.java

```diff
@@ -219,6 +218,150 @@ public void setUserInputType() {
+          }
+          this.rows = rows;
+          this.valid = true;
+        } catch (InputMismatchException e) {
+          System.out.println(e.getMessage());
+        } catch (Exception e) {
+          System.out.println("Invalid Input.");
+        }
+      }
+
+      this.valid = false; // Reset flag for next input
+
+      // Number of Columns
+      while (!this.valid) {
+        try {
+          if (columns <= 0) {
+            throw new InputMismatchException("Invalid Input: Please enter a valid number of columns.");
+          }
+          this.columns = columns;
+          this.valid = true;
+        } catch (InputMismatchException e) {
+          System.out.println(e.getMessage());
+        } catch (Exception e) {
+          System.out.println("Invalid Input.");
+        }
+      }
+
+    } else {
+      // User Input Mode
+
+      // Number of Matrices
+      while (!this.valid) {
+        try {
+          if (numberOfMatrices <= 0) {
+            throw new InputMismatchException("Invalid Input: Please enter a valid number of matrices.");
+          }
+          this.numberOfMatrices = numberOfMatrices;
+          this.valid = true;
+        } catch (InputMismatchException e) {
+          System.out.println(e.getMessage());
+        } catch (Exception e) {
+          System.out.println("Invalid Input.");
+        }
+      }
+
+      this.valid = false; // Reset flag for next input
+
+      // Number of Rows
+      while (!this.valid) {
+        try {
+          if (rows <= 0) {
+            throw new InputMismatchException("Invalid Input: Please enter a valid number of rows.");
+          }
+          this.rows = rows;
+          this.valid = true;
+        } catch (InputMismatchException e) {
+          System.out.println(e.getMessage());
+        } catch (Exception e) {
+          System.out.println("Invalid Input.");
+        }
+      }
+
+      this.valid = false; // Reset flag for next input
+
+      // Number of Columns
+      while (!this.valid) {
+        try {
+          if (columns <= 0) {
+            throw new InputMismatchException("Invalid Input: Please enter a valid number of columns.");
+          }
+          this.columns = columns;
+          this.valid = true;
+        } catch (InputMismatchException e) {
+          System.out.println(e.getMessage());
+        } catch (Exception e) {
+          System.out.println("Invalid Input.");
+        }
+      }
+
+      this.valid = false; // Reset flag for next input
+
+      // Multiply Matrices (Yes or No)
+      while (!this.valid) {
+        try {
+          if (multiply != 0 && multiply != 1) {
+            throw new InputMismatchException("Invalid Input: Please enter 0 or 1.");
+          }
+          this.multiply = multiply;
+          this.valid = true;
+        } catch (InputMismatchException e) {
+          System.out.println(e.getMessage());
+        } catch (Exception e) {
+          System.out.println("Invalid Input.");
+        }
+      }
+    }
+  }
+
```

Comment:
why are all of the inputs put into one method here?

Reply:
So The test can check to see if the data validation is ensuring that only valid data passes through the input config without having to manually input each value using scanner.